myserver.cpp

```diff
@@ -0,0 +1,626 @@
+
+    connect(socket, SIGNAL(readyRead()),this,SLOT(sockReady()));
+    connect(socket, SIGNAL(disconnected()),this,SLOT(sockDisc()));
+
+    qDebug()<<pisocketDescriptor<<"Client connected";
+
+    socket->write("{\"type\":\"connect\",\"status\":\"yes\"}");
+    qDebug()<<"Send client connect status - YES";
+}
+
+void CMyServer::sockReady(){
+    Data = socket->readAll();
+    qDebug()<<"Data: "<<Data;
+    std::vector <std::string> msg;
+
+    nlohmann::json j;
+    j = nlohmann::json::parse(QString(Data).toStdString());
+
+    for (nlohmann::json::iterator it = j.begin(); it != j.end(); ++it) {
+            msg.push_back(it.value());
+    }
+
+        if(msg.at(0)=="AddUser"){
+            CMyServer::m_AddUser(msg);
+        }
+        else if(msg.at(0)=="GetUserId"){
+            CMyServer::m_GetUserId(msg);
+        }
+        else if(msg.at(0)=="GetFriendList"){
+            CMyServer::m_GetFriendList(msg);
+        }
+        else if(msg.at(0)=="AddFriend"){
+            CMyServer::m_AddFriend(msg);
+        }
+        else if(msg.at(0)=="DeleteFriend"){
+            CMyServer::m_DeleteFriend(msg);
+        }
+        else if(msg.at(0)=="GetFriendRequestsList"){
+            CMyServer::m_GetFriendRequestsList(msg);
+        }
+        else if(msg.at(0)=="AddFriendRequest"){
+            CMyServer::m_AddFriendRequest(msg);
+        }
+        else if(msg.at(0)=="DeleteFriendRequest"){
+            CMyServer::m_DeleteFriendRequest(msg);
+        }
+        else if(msg.at(0)=="CreateCategory"){
+            CMyServer::m_CreateCategory(msg);
+        }
+        else if(msg.at(0)=="GetCategory"){
+            CMyServer::m_GetCategory(msg);
+        }
+        else if(msg.at(0)=="DeleteCategory"){
+            CMyServer::m_DeleteCategory(msg);
+        }
+        else if(msg.at(0)=="ChangeCategoryName"){
+            CMyServer::m_ChangeCategoryName(msg);
+        }
+        else if(msg.at(0)=="AddNote"){
+            CMyServer::m_AddNote(msg);
+        }
+        else if(msg.at(0)=="GetNote"){
+            CMyServer::m_GetNote(msg);
+        }
+        else if(msg.at(0)=="ChangeCategoryId"){
+            CMyServer::m_ChangeCategoryId(msg);
+        }
+        else if(msg.at(0)=="SetHeader"){
+            CMyServer::m_SetHeader(msg);
+        }
+        else if(msg.at(0)=="SetText"){
+            CMyServer::m_SetText(msg);
+        }
+        else if(msg.at(0)=="AddFriendAccess"){
+            CMyServer::m_AddFriendAccess(msg);
+        }
+        else if(msg.at(0)=="CheckFriendAccess"){
+            CMyServer::m_CheckFriendAccess(msg);
+        }
+        else if(msg.at(0)=="DeleteFriendAccess"){
+            CMyServer::m_DeleteFriendAccess(msg);
+        }
+        else{
+            bytes = "Unclear command";
+            CMyServer::m_socketWrite(bytes);
+         }
+}
+void CMyServer::m_AddUser(std::vector <std::string> &msg){
+    int iRtCode = IDataBase::AddUser(msg[1], msg[2], iUserId);
```

Comment:
use at()
correct this in all the files.

Reply:
Fixed it